skimage/transform/_distance_transform.pyx

```diff
@@ -70,23 +70,24 @@ def _generalized_distance_transform_1d_euclidean(double[:] arr, double[:] cost_a
     domains[1] = INFINITY
     centers[0] = start
     
-    for i in range(start+1,length):
-        intersection = euclidean_meet(i,centers[rightmost],cost_arr)
-        while intersection <= domains[rightmost] or domains[rightmost]==INFINITY and rightmost>start:
-            rightmost-=1
+    with nogil:
```

Comment:
did it compile? cool!

Reply:
I think you should move this way to the top, you don't seem to be touching pything anywhere in your data sanitization either.

Reply:
just below the declaration i think. It seems ok now. maybe i was looking at some old version :/